packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -48,15 +55,22 @@ class CodeSizeController {
 
     final diffMap = await buildComparisonTreemap(oldInputJson, newInputJson);
     diffMap['n'] = 'Root';
-    final newRoot = generateTree(diffMap, showDiff: true);
+    final newRoot = generateTree(
+      diffMap,
+      diffTreeType: diffTreeType,
+      showDiff: true,
+    );
 
     changeRoot(newRoot);
   }
 
   /// Builds a tree with [TreemapNode] from [treeJson] which represents
   /// the hierarchical structure of the tree.
-  TreemapNode generateTree(Map<String, dynamic> treeJson,
-      {bool showDiff = false}) {
+  TreemapNode generateTree(
+    Map<String, dynamic> treeJson, {
+    DiffTreeType diffTreeType = DiffTreeType.combined,
+    bool showDiff = false,
+  }) {
```

Comment:
it seems like we should have two different methods: `generateTree` and `generateDiffTree` rather than putting a bunch of diff-specific code into the general treemap code. For example, when creating a regular treemap to view a snaphsot, the diffTreeType node is irrelevant.